### PR TITLE
#12615: Add queue_id output tensor to slice op, concat_bw

### PR DESCRIPTION
**Pull request by KalaivaniMCW**
### Ticket
Link to Github Issue #12615

### Problem description
Add queue_id and optional output tensor to slice op, concat_bw

### What's changed
Add queue_id and optional output tensor to slice op, concat_bw

### Checklist
- [x] Post commit CI https://github.com/tenstorrent/tt-metal/actions/runs/10902612202
https://github.com/tenstorrent/tt-metal/actions/runs/10937632933
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/reduction/prod/prod.cpp [hunk 1]
```diff
@@ -115,7 +115,7 @@ Tensor ProdOperation::invoke(const Tensor& input_a, bool all_dimensions, int64_t
         tt::tt_metal::LegacyShape input_shape = input_a.get_legacy_shape();
         std::vector<uint32_t> start_index = {0, 0, 0, 0};
         std::vector<uint32_t> end_index = {input_shape[0] - 1, input_shape[1] - 1, 0, input_shape[2] - 1};
-        Tensor new_unpad_tensor = ttnn::slice(0, required, start_index, end_index, std::nullopt, std::nullopt);
+        Tensor new_unpad_tensor = ttnn::slice(0, required, start_index, end_index, std::nullopt, std::nullopt, std::nullopt);
```
Comment (KalaivaniMCW) [inline ttnn/cpp/ttnn/operations/reduction/prod/prod.cpp:118]: can these 0 be replaced with DefaultQueueId ?

**Comment by tarafdarTT** [comment]
is it just adding queue_id or optional_tensor as well?

**Comment by KalaivaniMCW** [comment]
> is it just adding queue_id or optional_tensor as well?

@tarafdarTT Need both queue_id and optional_tensor.

**Comment by KalaivaniMCW** [comment]
@tarafdarTT would like a review from the DM team.

**Review by sjameelTT — APPROVED** [review]

**Review by eyonland — APPROVED** [review]
I would recommend to update the the code to handle the defaults, so that we don't need to do this...
ttnn::slice(0, result, start_index, end_index, std::nullopt, std::nullopt, std::nullopt)
and instead we can just have
ttnn::slice(result, start_index, end_index)

**Comment by KalaivaniMCW** [comment]
> I would recommend to update the the code to handle the defaults, so that we don't need to do this... ttnn::slice(0, result, start_index, end_index, std::nullopt, std::nullopt, std::nullopt) and instead we can just have ttnn::slice(result, start_index, end_index)

sure @eyonland will do. updating the struct in `slice.hpp` with default values will achieve that.

**Pull request merged**